### PR TITLE
Test generating a PCA on TOB-WGS data

### PR DESCRIPTION
**Pull request by KatalinaBobowik**
Test generating a PCA on TOB-WGS data that have been filtered to variants present in the SNP-chip data.

### CODE REVIEW

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    snp_chip = hl.read_matrix_table(SNP_CHIP).key_rows_by('locus', 'alleles')
+
+    # filter to loci that are contained in snp-chip data after densifying
+    tob_wgs = hl.experimental.densify(tob_wgs)
+    tob_wgs = tob_wgs.semi_join_rows(snp_chip.rows())
+    tob_wgs = tob_wgs.cache()
+    print(tob_wgs.count_rows())
+    tob_wgs_path = f'{output}/tob_wgs_filtered_snp_chip.mt'
+    tob_wgs = tob_wgs.repartition(100, shuffle=False)
+    tob_wgs.write(tob_wgs_path)
```
Comment (lgruen) [inline scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py:31]: How about forming the `union` of sample columns? That'll give you the row restriction immediately and you can run a joint PCA afterwards.

**Review by lgruen — CHANGES_REQUESTED** [review]

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py (hunk 1 above)
Comment (KatalinaBobowik) [reply scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py:31]: Would it be better to project the SNP-chip samples onto the tob-wgs pca space? I guess the question is whether we want to look at the SNP-chip data in relation to the tob-wgs sample mean, or if we want to allow the snp-chip samples to skew the mean. I think the datasets should be similar enough that it won't actually matter either way, but my first thought was projecting the snp-chip data back onto the tob-wgs pca space

Comment (lgruen) [reply scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py:31]: Wasn't that what you ran previously? If you use loci that aren't present in the SNP-chip data, I think you'll always get the "shrinkage" effect that your previous plot showed.

**Review by lgruen — CHANGES_REQUESTED** [review]
Added a comment to our thread.

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py (hunk 1 above)
Comment (KatalinaBobowik) [reply scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py:31]: The previous plot was without any filtering. This is filtering the tob-wgs matrix table to variants that are present in the snp-chip data (and I would also have to filter the snp-chip data to variants that are present in the tob-wgs matrix table). I'm happy to do a union of the sample columns, if that seems more efficient. I don't think the effect will be huge either way

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+"""
+Generate PCA on combined TOB-WGS/SNP-chip data.
+"""
+
+import click
+import hail as hl
+import pandas as pd
+
+TOB_WGS = 'gs://cpg-tob-wgs-main/mt/v2-raw.mt/'
```
Comment (lgruen) [inline scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py:9]: Don't we have a newer version with more samples?

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    hl.init(default_reference='GRCh38')
+
+    tob_wgs = hl.read_matrix_table(TOB_WGS).key_rows_by('locus', 'alleles')
+    snp_chip = hl.read_matrix_table(SNP_CHIP).key_rows_by('locus', 'alleles')
+
+    # filter to loci that are contained in snp-chip data after densifying
+    tob_wgs = hl.experimental.densify(tob_wgs)
+    tob_wgs = tob_wgs.select_entries(tob_wgs.GT).select_cols()
+    snp_chip = snp_chip.select_entries(snp_chip.GT).select_cols()
```
Comment (lgruen) [inline scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py:26]:
```suggestion
    tob_wgs = tob_wgs.select_cols().select_entries(tob_wgs.GT)
    snp_chip = snp_chip.select_cols().select_entries(snp_chip.GT)
```

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    tob_combined = tob_wgs.union_cols(snp_chip)
+    tob_combined = tob_combined.cache()
+    print(tob_combined.count_rows())
+    tob_combined_path = f'{output}/tob_wgs_snp_chip_combined.mt'
+    tob_combined = tob_wgs.repartition(1000, shuffle=False)
+    tob_combined.write(tob_combined_path)
```
Comment (lgruen) [inline scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py:32]: Let's maybe try without repartitioning / writing? :)

```suggestion
```

**Review by lgruen — APPROVED** [review]

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py (hunk 2 above)
Comment (KatalinaBobowik) [reply scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py]: Yes, we do! Thanks for pointing me to this. I actually had no idea there was a new one.

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py (hunk 3 above)
Comment (KatalinaBobowik) [reply scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py]: Just out of curiosity, does the ordering matter here? Or is it about readability?

File: scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py (hunk 4 above)
Comment (KatalinaBobowik) [reply scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py]: I am not against that :D

**Pull request merged**